src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
 public class FileWork {
+    private static final String SPECIFIED_LETTER = "w";
+    private static final String DIVIDER = " ";
+    private static final String REGEX = "[\\W]+";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            StringBuilder builder = new StringBuilder();
+            int value = bufferedReader.read();
+            while (value != -1) {
+                builder.append((char) value);
+                value = bufferedReader.read();
+            }
+            String[] results = builder.toString().split(REGEX);
+            StringBuilder filteredWords = new StringBuilder();
+            for (String result : results) {
+                String lowerCase = result.toLowerCase();
+                if (lowerCase.startsWith(SPECIFIED_LETTER)) {
+                    filteredWords.append(lowerCase).append(DIVIDER);
+                }
+            }
+            if (filteredWords.length() == 0) {
+                return new String[0];
+            }
+            String[] finalResults = filteredWords.toString().split(DIVIDER);
+            Arrays.sort(finalResults);
+            bufferedReader.close();
+            return finalResults;
+        } catch (IOException e) {
+            throw new RuntimeException("Can not read a file", e);
```

Comment:
and what about this one? https://github.com/mate-academy/jv-read-from-file/pull/207/files#r606968005